### PR TITLE
[FRAME] Message Queue use proper overweight limit

### PR DESCRIPTION
**Pull request by ggwpez**
Changes:
- Use a sensible limit for the overweight-cutoff of a single messages instead of the full configured `ServiceWeight`.
- Add/Update tests

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3961911

File: substrate/frame/message-queue/src/lib.rs [hunk 1]
```diff
@@ -759,6 +760,47 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// The maximal weight that a single message.
```
Comment (ggwpez) [inline substrate/frame/message-queue/src/lib.rs:763]:
```suggestion
	/// The maximal weight that a single message can consume.
```

**Review by KiChjang — APPROVED** [review]

**Review by liamaharon — APPROVED** [review]

File: substrate/frame/message-queue/src/lib.rs [hunk 2]
```diff
@@ -759,6 +760,47 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// The maximal weight that a single message can consume.
+	///
+	/// Any message using more than this will be marked as permanently overweight and not
+	/// automatically re-attempted. Returns `None` if the servicing of a message cannot begin.
+	/// `Some(0)` means that only messages with no weight may be served.
+	fn overweight_limit(limit: Weight) -> Option<Weight> {
```
Comment (franciscoaguirre) [inline substrate/frame/message-queue/src/lib.rs:768]: Maybe it's just me, but I'd find it less confusing if `overweight_limit` became `max_message_weight`, so we don't have two limits. WDYT?

**Review by franciscoaguirre — APPROVED** [review]

File: substrate/frame/message-queue/src/lib.rs [hunk 3]
```diff
@@ -759,6 +760,47 @@ impl<T: Config> Pallet<T> {
...
+		limit.checked_sub(&Self::single_msg_overhead())
+	}
+
+	/// The overhead of servicing a single message.
+	fn single_msg_overhead() -> Weight {
```
[the remark this answers is not in the document]
Comment (ggwpez) [reply substrate/frame/message-queue/src/lib.rs:773]: Yea, const trait functions would be very nice.

File: substrate/frame/message-queue/src/integration_test.rs [hunk 4]
```diff
@@ -17,6 +17,13 @@
 
 //! Stress tests pallet-message-queue. Defines its own runtime config to use larger constants for
 //! `HeapSize` and `MaxStale`.
+//!
+//! The tests in this file are not run per default, since they are quite slow in debug mode. You can
+//! run them manually while debug output van be enabled with `RUST_LOG=info`.
```
[the remark this answers is not in the document]
Comment (ggwpez) [reply substrate/frame/message-queue/src/integration_test.rs:22]: I re-wrote it a bit.

File: substrate/frame/message-queue/src/lib.rs (hunk 2 above)
Comment (ggwpez) [reply substrate/frame/message-queue/src/lib.rs:768]: Yea agreed. It sounds confusing :see_no_evil:

File: substrate/frame/message-queue/src/integration_test.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ggwpez) [reply substrate/frame/message-queue/src/integration_test.rs:22]: Its actually `info`, but i changed the sentence to just put it into the command as well :+1:

**Pull request merged**